### PR TITLE
Donation config corrections

### PR DESCRIPTION
**Pull request by adammeyer**
This are corrections from QA.

1) In the AMOUNT section – the last option is for "other amount" – for some reason it is getting pre-populated with whichever amount is being selected (for example "50"). If you try to change it, weird up/down arrows show up and make it feel like you have to click one at a time to increase the amount. See how United Way handles their amounts: https://donate.unitedway.org/get-involved/ways-to-give/donate – that's the standard behavior we want.
2) GIFT FREQUENCY - change "One Time" to "Single Gift"
3) OPTIONAL COMMENTS - drop the word "Comments" from the title
4) "Special Handling instructions for Processing the Gift" should read "Special Handling Instructions for Processing This Gift"

### CODE REVIEW

File: src/app/productConfig/productConfigModal.tpl.html [hunk 1]
```diff
@@ -35,20 +35,21 @@ <h4 class="panel-title border-bottom-small" translate>
...
-              <label class="btn btn-radio" ng-click="$ctrl.itemConfig.amount = 250" ng-class="{'active': $ctrl.itemConfig.amount === 250}"><span class="number">$250</span></label>
-              <label class="btn btn-radio" ng-click="$ctrl.itemConfig.amount = 500" ng-class="{'active': $ctrl.itemConfig.amount === 500}"><span class="number">$500</span></label>
-              <label class="btn btn-radio" ng-click="$ctrl.itemConfig.amount = 1000" ng-class="{'active': $ctrl.itemConfig.amount === 1000}"><span class="number">$1000</span></label>
-              <label class="btn btn-radio" ng-click="$ctrl.itemConfig.amount = 5000" ng-class="{'active': $ctrl.itemConfig.amount === 5000}"><span class="number">$5000</span></label>
+              <label class="btn btn-radio" ng-click="$ctrl.changeAmount(a)" ng-class="{'active': $ctrl.itemConfig.amount === a}" ng-repeat-start="a in $ctrl.selectableAmounts">
+                <span class="number">${{a}}</span>
+              </label>
+              <span ng-repeat-end></span>
```
Comment (OzzieOrca) [inline src/app/productConfig/productConfigModal.tpl.html]: How come normal `ng-repeat` doesn't work here? Is that `span` required between each label?

File: src/app/productConfig/productConfigModal.tpl.html [hunk 2]
```diff
@@ -35,20 +35,21 @@ <h4 class="panel-title border-bottom-small" translate>
...
+                  <input class="form-control form-control-subtle number" type="text" name="amount"
+                         ng-model="$ctrl.customAmount"
+                         ng-change="$ctrl.itemConfig.amount = $ctrl.customAmount"
+                         ng-pattern="/^[1-9]\d*(\.\d+)?$/">
                 </div>
               </label>
               <div role="alert" ng-messages="$ctrl.itemConfigForm.amount.$error" ng-if="($ctrl.itemConfigForm.amount | showErrors)">
                 <div class="help-block" ng-message="required" translate>Amount must be at least $1</div>
-                <div class="help-block" ng-message="min" translate>Amount must be at least $1</div>
+                <div class="help-block" ng-message="pattern" translate>Amount must be at least $1</div>
```
Comment (OzzieOrca) [inline src/app/productConfig/productConfigModal.tpl.html]: This error could be misleading if someone types `1.` Maybe adding something about it being a valid number would be good.

File: src/app/productConfig/productConfigModal.tpl.html (hunk 1 above)
Comment (adammeyer) [inline src/app/productConfig/productConfigModal.tpl.html]: A space is, but normal ng-repeat puts the labels too close together.

File: src/app/productConfig/productConfigModal.tpl.html [hunk 3]
```diff
@@ -63,7 +64,10 @@ <h4 class="panel-title border-bottom-small mt" translate>
                      ng-repeat-start="f in $ctrl.productData.frequencies | orderBy: $ctrl.frequencyOrder"
                      ng-class="{'active': !f.selectAction}"
                      ng-click="$ctrl.changeFrequency(f.selectAction)">
-                <span class="giftsum-about giftsum-title">{{f.display}}</span>
+                <span class="giftsum-about giftsum-title" ng-switch="f.name">
+                  <span ng-switch-when="NA" translate>Single Gift</span>
+                  <span ng-switch-default>{{f.display}}</span>
+                </span>
               </label>
               <span ng-repeat-end></span>
```
Comment (OzzieOrca) [inline src/app/productConfig/productConfigModal.tpl.html]: Using `ng-repeat-start` and `ng-repeat-end` is here too...

File: src/app/productConfig/productConfigModal.tpl.html [hunk 4]
```diff
@@ -126,7 +130,7 @@ <h4 class="panel-title border-bottom-small mt" translate>
                         style="display:block;"></textarea>
               </li>
               <li>
-                <a class="commentLink" href="" ng-click="showDSComments = !showDSComments" translate>Special Handling Instructions for Processing the Gift</a>
+                <a class="commentLink" href="" ng-click="showDSComments = !showDSComments" translate>Special Handling Instructions for Processing This Gift</a>
```
Comment (OzzieOrca) [inline src/app/productConfig/productConfigModal.tpl.html]: Does the `This` need to be capitalized?

Comment (adammeyer) [inline src/app/productConfig/productConfigModal.tpl.html]: I copied it from the task.  I'll check.

File: src/app/productConfig/productConfigModal.tpl.html (hunk 1 above)
Comment (OzzieOrca) [inline src/app/productConfig/productConfigModal.tpl.html]: Oh interesting. I guess ideally that would be done in CSS

**Comment by OzzieOrca** [comment]
Besides comments, 👍

File: src/app/productConfig/productConfigModal.tpl.html (hunk 1 above)
Comment (adammeyer) [inline src/app/productConfig/productConfigModal.tpl.html]: idk.  What they have is valid html.  ng-repeat just removes the spacing between each element.  I think this is the best way to resolve that.

**Pull request merged**